packages/@aws-cdk/aws-servicecatalogappregistry/lib/application-associator.ts

```diff
@@ -56,17 +137,21 @@ export class ApplicationAssociator extends Construct {
   constructor(scope: cdk.App, id: string, props: ApplicationAssociatorProps) {
     super(scope, id);
 
-    const applicationStack = new cdk.Stack(scope, 'ApplicationAssociatorStack', props.stackProps);
+    if (props.applications.length != 1) {
+      throw new Error('Please provide either ARN or application name.');
```

Comment:
This error message is not expressed in terms of API elements the user is familiar with, and is even inaccurate if  `length == 2`.

```
Please pass exactly 1 instance of ApplicationTarget.createApplication(), ApplicationTarget.existingApplicationFromArn(), etc, into the 'application' property
```